src/pkgcore/ebuild/processor.py

```diff
@@ -691,7 +691,7 @@ def _generate_env_str(self, env_dict):
         for key, val in sorted(env_dict.items()):
             if key in self._readonly_vars:
                 continue
-            if not key[0].isalpha():
+            if not key[0].isalpha() and not key.startswith("_"):
                 raise KeyError(f"{key}: bash doesn't allow digits as the first char")
```

Comment:
error message needs updating.

Reply:
Thanks. I hadn't got back to this as I got distracted & wanted to make the change mgorny suggested because it was unclear if we should be allowing _more_ than just alpha + _, but I don't mind just going with your change.